src/main/java/nl/hsac/fitnesse/fixture/slim/web/BrowserTest.java

```diff
@@ -1539,6 +1539,52 @@ public boolean isVisibleOnPageIn(String place, String container) {
         return isVisibleImpl(place, container, false);
     }
 
+    /**
+     * Determines whether element is not visible (or disappears within the specified timeout)
+     * @param place element to check
+     * @return true if the element is not displayed (anymore)
+     */
+    @WaitUntil(TimeoutPolicy.RETURN_FALSE)
+    public boolean notVisible(String place) {
+        return notVisibleIn(place, null);
+    }
+
+    /**
+     * Determines whether element is not visible (or disappears within the specified timeout)
+     * @param place element to check.
+     * @param container parent of place.
+     * @return true if the element is not displayed (anymore)
+     */
+    @WaitUntil(TimeoutPolicy.RETURN_FALSE)
+    public boolean notVisibleIn(String place, String container) {
+        return notVisibleImpl(place, container, true);
+    }
+
+    /**
+     * Determines whether element is not on the page (or disappears within the specified timeout)
+     * @param place element to check.
+     * @return true if element is not on the page (anymore).
+     */
+    @WaitUntil(TimeoutPolicy.RETURN_FALSE)
+    public boolean notVisibleOnPage(String place) {
+        return notVisibleOnPageIn(place, null);
+    }
+
+    /**
+     * Determines whether element is not on the page (or disappears within the specified timeout)
+     * @param place element to check.
+     * @param container parent of place.
+     * @return true if the element is not on teh page (anymore)
+     */
+    @WaitUntil(TimeoutPolicy.RETURN_FALSE)
+    public boolean notVisibleOnPageIn(String place, String container) {
+        return notVisibleImpl(place, container, false);
+    }
+
+    protected boolean notVisibleImpl(String place, String container, boolean checkOnScreen) {
+        return waitDriver().until(d -> !isVisibleImpl(place, container, checkOnScreen));
```

Comment:
Is the explicit use of `waitDriver()` required? Would the `@WaitUntil` not take care of repeating the wait?
Your new test passes if I change the line to `return !isVisibleImpl(place, container, checkOnScreen)`...

Reply:
Yes, you're right. Keep it simple :)
That also removes the need for the extra impl.